src/devices/bus/hp_dio/human_interface.cpp

```diff
@@ -0,0 +1,304 @@
+	MCFG_IEEE488_DAV_CALLBACK(WRITELINE(m_tms9914, tms9914_device, dav_w))
+	MCFG_IEEE488_NRFD_CALLBACK(WRITELINE(m_tms9914, tms9914_device, nrfd_w))
+	MCFG_IEEE488_NDAC_CALLBACK(WRITELINE(m_tms9914, tms9914_device, ndac_w))
+	MCFG_IEEE488_IFC_CALLBACK(WRITELINE(m_tms9914, tms9914_device, ifc_w))
+	MCFG_IEEE488_SRQ_CALLBACK(WRITELINE(m_tms9914, tms9914_device, srq_w))
+	MCFG_IEEE488_ATN_CALLBACK(WRITELINE(m_tms9914, tms9914_device, atn_w))
+	MCFG_IEEE488_REN_CALLBACK(WRITELINE(m_tms9914, tms9914_device, ren_w))
+MACHINE_CONFIG_END
+
+ROM_START(human_interface)
+	ROM_REGION(0x800 , "iocpu" , 0)
+	ROM_LOAD( "1820-4784.bin", 0x000000, 0x000800, CRC(e929044a) SHA1(90849a10bdb8c6e38e73ce027c9c0ad8b3956b1b))
+ROM_END
+
+const tiny_rom_entry *human_interface_device::device_rom_region() const
+{
+	return ROM_NAME(human_interface);
+}
+
+void human_interface_device::iocpu_map(address_map& map)
+{
+	map(0x0000, 0x07ff).rom().region("iocpu", 0);
+}
+void human_interface_device::map(address_map& map)
+{
+	map(0x00420000, 0x00420003).mirror(0x0000fffc).rw(m_iocpu, FUNC(upi41_cpu_device::upi41_master_r), FUNC(upi41_cpu_device::upi41_master_w)).umask32(0x00ff00ff);
+	map(0x00470000, 0x0047001f).mirror(0x0000ffe0).rw(FUNC(human_interface_device::gpib_r), FUNC(human_interface_device::gpib_w)).umask16(0x00ff);
+
+}
+
+human_interface_device::human_interface_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	human_interface_device(mconfig, HPDIO_HUMAN_INTERFACE, tag, owner, clock)
+{
+}
+
+human_interface_device::human_interface_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_dio16_card_interface(mconfig, *this),
+	m_iocpu(*this, "iocpu"),
+	m_mlc(*this, "mlc"),
+	m_sound(*this, "sn76494"),
+	m_tms9914(*this, "tms9914"),
+	m_rtc(*this, "rtc")
+{
+}
+
+void human_interface_device::device_start()
+{
+	program_space()->install_readwrite_handler(0x420000, 0x420003, 0x0003, 0xfffc, 0,
+		read8_delegate(FUNC(upi41_cpu_device::upi41_master_r), &(*m_iocpu)),
+		write8_delegate(FUNC(upi41_cpu_device::upi41_master_w), &(*m_iocpu)), 0x00ff00ff);
+
+	program_space()->install_readwrite_handler(0x470000, 0x47001f, 0x1f, 0xffe0, 0,
+		read8_delegate(FUNC(human_interface_device::gpib_r), this),
+		write8_delegate(FUNC(human_interface_device::gpib_w), this), 0x00ff00ff);
+
+	save_item(NAME(m_hil_read));
+	save_item(NAME(m_kbd_nmi));
+	save_item(NAME(gpib_irq_line));
+	save_item(NAME(m_old_latch_enable));
+	save_item(NAME(m_hil_data));
+	save_item(NAME(m_latch_data));
+	save_item(NAME(m_rtc_data));
+}
+
+void human_interface_device::device_reset()
+{
+	m_rtc->cs1_w(ASSERT_LINE);
+	m_rtc->cs2_w(CLEAR_LINE);
+	m_rtc->write_w(CLEAR_LINE);
+	m_rtc->read_w(CLEAR_LINE);
+	m_rtc->cs2_w(CLEAR_LINE);
+	m_iocpu->reset();
+}
+
+WRITE_LINE_MEMBER(human_interface_device::reset_in)
+{
+	if (state)
+		device_reset();
```

Comment:
You might also want to `m_iocpu->set_input_line(INPUT_LINE_RESET, state);` if the I/O CPU is going to be held in reset while the line is asserted.